src/main/java/sg/ncl/MainController.java

```diff
@@ -2105,113 +2101,118 @@ public String data(Model model, HttpSession session) throws Exception {
-//    	System.out.println(dataset.getDatasetId());
-//
-//    	if (origDataset.updateName(editedDatasetName) == true) {
-//    		redirectAttributes.addFlashAttribute("editName", "success");
-//    	}
-//
-//    	if (origDataset.updateDescription(editedDatasetDesc) == true) {
-//    		redirectAttributes.addFlashAttribute("editDesc", "success");
-//    	}
-//
-//    	if (origDataset.updateLicense(editedDatasetLicense) == true) {
-//    		redirectAttributes.addFlashAttribute("editLicense", "success");
-//    	}
-//
-//    	if (origDataset.updatePublic(editedDatasetPublic) == true) {
-//    		redirectAttributes.addFlashAttribute("editPublic", "success");
-//    	}
-//
-//    	if (origDataset.updateAuthorization(editedDatasetIsRequiredAuthorization) == true) {
-//    		redirectAttributes.addFlashAttribute("editIsRequiredAuthorization", "success");
-//    	}
-//
-//    	datasetManager.updateDatasetDetails(origDataset);
-//
-//    	return "redirect:/data/edit/{datasetId}";
-//    }
+        try {
+            if (RestUtil.isError(response.getStatusCode())) {
+                MyErrorResource error = objectMapper.readValue(dataResponseBody, MyErrorResource.class);
+                ExceptionState exceptionState = ExceptionState.parseExceptionState(error.getError());
 
-//    @RequestMapping("/data/remove_dataset/{datasetId}")
-//    public String removeDataset(@PathVariable Integer datasetId) {
-//    	datasetManager.removeDataset(datasetId);
-//    	return "redirect:/data";
-//    }
+                switch (exceptionState) {
+                    case DATASET_NAME_IN_USE_EXCEPTION:
+                        model.addAttribute("message", "Error(s):<ul><li>dataset name already exists</li></ul>");
+                        break;
+                    case FORBIDDEN_EXCEPTION:
+                        model.addAttribute("message", "Error(s):<ul><li>saving dataset forbidden</li></ul>");
+                        break;
+                }
+
+                return "data_contribute";
+            }
+        } catch (IOException e) {
+            throw new WebServiceRuntimeException(e.getMessage());
+        }
+
+        return "redirect:/data";
+    }
+
+    @RequestMapping("/data/remove/{id}")
+    public String removeDataset(@PathVariable String id, RedirectAttributes redirectAttributes) throws Exception {
+        HttpEntity<String> request = createHttpEntityHeaderOnly();
+        restTemplate.setErrorHandler(new MyResponseErrorHandler());
+        ResponseEntity response = restTemplate.exchange(properties.getDataset(id), HttpMethod.DELETE, request, String.class);
+        String dataResponseBody = response.getBody().toString();
+
+        try {
+            if (RestUtil.isError(response.getStatusCode())) {
+                MyErrorResource error = objectMapper.readValue(dataResponseBody, MyErrorResource.class);
+                ExceptionState exceptionState = ExceptionState.parseExceptionState(error.getError());
+
+                switch (exceptionState) {
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Add a default case to this switch. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3ASwitchLastCaseIsDefaultCheck)
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Replace this "switch" statement by "if" statements to increase readability. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1301)